crates/katana/rpc/rpc/src/starknet.rs

```diff
@@ -657,4 +665,35 @@ impl StarknetApiServer for StarknetApi {
         })
         .await
     }
+
+    async fn simulate_transactions(
+        &self,
+        request: SimulateTransactionsRequest,
+    ) -> RpcResult<Vec<SimulatedTransaction>> {
+        self.on_cpu_blocking_task(move |this| {
+            let charge_fee = !request.simulation_flags.contains(&SimulationFlag::SkipFeeCharge);
+            let validate = !request.simulation_flags.contains(&SimulationFlag::SkipValidate);
+            let executables = request
+                .try_into_executables_with_hash(this.inner.sequencer.chain_id())
+                .map_err(|_| StarknetApiError::ContractError {
+                    revert_error: "Transaction conversion error".to_string(),
+                })?;
+            let simulated_transactions = executables
+                .into_iter()
+                .map(|exec| {
+                    this.inner.sequencer.simulate_transaction(
+                        exec,
```

Comment:
as mentioned above regarding passing the whole vec instead, lets pass the whole vector directly starting from here all the way until the util function.